interfaces/builtin/login_control.go

```diff
@@ -0,0 +1,67 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const loginControlConnectedPlugAppArmor = `
+# Description: Can control the login system service
+
+#include <abstractions/dbus-strict>
+
+# Give full access to the login service. This allows a client
+# to control everything (system reboot, shutdown, suspend, ...)
```

Comment:
As written, I think this would need to be renamed 'systemd-login-control' since it is dealing strictly with systemd-logind as listed here: https://freedesktop.org/wiki/Software/systemd/logind/. 'login-control' conveys (to me anyway) things like controlling PAM, the shadow file (eg, hash algorithm, password age, etc), changing GECOS, login shells, login.defs, etc, etc.

This interface grants the entire login1 API, which I'm not sure we want. For one, it overlaps with the shutdown interface, but it also includes disparate APIs as well as ones that shouldn't be used by clients, such as CreateSession() and ReleaseSession() (per the above website). Furthermore, systemd-logind has extensive polkit integration but the core snap doesn't have polkitd (yet).

What are the specific problems you are trying to solve with this interface? Perhaps this PR can be refined/renamed to more specifically target your use cases. I'm going to defer a full review for now.